pcsx2/ImGui/FullscreenUI.cpp

```diff
@@ -4297,25 +4325,123 @@ void FullscreenUI::DrawPatchesOrCheatsSettingsPage(bool cheats)
 		}
 	}
 
-	for (const Patch::PatchInfo& pi : patch_list)
+
+
+	struct PatchGroup
 	{
-		const auto enable_it = std::find(enable_list.begin(), enable_list.end(), pi.name);
+		bool bUncategorizedContents = false;
+		std::vector<std::string>		headers;			//	Total Headers
+		std::vector<Patch::PatchInfo>	pInfoList;			//	All Patches
+	};
+	PatchGroup s_patch_group;
```

Comment:
You shouldn't be allocating memory every frame. Generate it once and cache it.

(yes, I know there's some other places which do std::string allocation, those will get removed once I get around to localizing this thing)

Reply:
I have moved this to another place. But since I cannot obtain the member names upon initialization this entire method is worthless